app/controllers/devise_token_auth/omniauth_callbacks_controller.rb

```diff
@@ -161,12 +161,11 @@ def create_auth_params
       @auth_params = {
         auth_token:     @token,
         client_id: @client_id,
-        uid:       @resource.uid,
-        provider: 'email',
         expiry:    @expiry,
         config:    @config
       }
-      @auth_params.merge!(@resource.build_auth_header(@token, @client_id, @provider_id, @provider))
+      hash = @resource.build_auth_header(@token, @client_id, @provider_id, @provider)
+      @auth_params.merge!(hash)
```

Comment:
I think you need only two values to be merged: `provider` and `uid`. So do not merge all fields from `auth_header`. Keep just what you need/
```ruby
@auth_param.merge!(hash.slice(:provider, :uid))
```